### PR TITLE
Feature/clear user data

### PR DESCRIPTION
**Pull request by dennis-montana**
- Added option to remove all customer data
- Disabled UrbanAirship code which broke the demo app.

### CODE REVIEW

File: live-engage.android.ts [hunk 1]
```diff
@@ -119,4 +119,25 @@ export class LiveEngage implements CommonLiveEngage {
...
+            return;
+        }
+
+        if (!this.isValidState()) {
+            return;
+        }
+
+        com.liveperson.messaging.MessagingFactory.getInstance().getController().resolveConversation(this.brandId, this.brandId);
+        const LogoutCallback: any = com.liveperson.infra.callbacks.LogoutLivePersonCallBack.extend({
+            onLogoutSucceed: () => { },
```
Comment (spike1292) [inline live-engage.android.ts]: think we should expose this call back as a promise

File: live-engage.android.ts [hunk 2]
```diff
@@ -119,4 +119,25 @@ export class LiveEngage implements CommonLiveEngage {
...
+    public killChat(): void {
+        if (!this.brandId || !this.appId) {
+            return;
+        }
+
+        if (!this.isValidState()) {
+            return;
+        }
+
+        com.liveperson.messaging.MessagingFactory.getInstance().getController().resolveConversation(this.brandId, this.brandId);
```
Comment (servehermans) [inline live-engage.android.ts]: Why supply two times the brandid parameter?

File: live-engage.android.ts [hunk 3]
```diff
@@ -119,4 +119,25 @@ export class LiveEngage implements CommonLiveEngage {
...
+        const LogoutCallback: any = com.liveperson.infra.callbacks.LogoutLivePersonCallBack.extend({
+            onLogoutSucceed: () => { },
+            onLogoutFailed: (err: any) => {
+                console.error(err);
+            }
+        });
+        const initProperties = new com.liveperson.infra.InitLivePersonProperties(this.brandId, this.appId, null);
+        const ui = new com.liveperson.infra.messaging_ui.MessagingUiInitData(initProperties, this.getSDKVersion());
+        com.liveperson.infra.messaging_ui.MessagingUIFactory.getInstance().logout(application.android.context, ui, new LogoutCallback());
```
Comment (servehermans) [inline live-engage.android.ts]: Logging out means the instance will not run anymore and has to be reinitialised next time. Is that the only way to get rid of the customer data or would be clearing the history good enough? Secondly, what happens to registered push notifications, are these unregistered as well?

File: live-engage.android.ts (hunk 2 above)
Comment (dennis-montana) [inline live-engage.android.ts]: This is needed for the SDK.

File: live-engage.android.ts (hunk 3 above)
Comment (dennis-montana) [inline live-engage.android.ts]: According to the documentation this is the correct way to remove all user data. This also unregisters for push notifications.

File: demo/app/activity.android.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import {setActivityCallbacks, AndroidActivityCallbacks} from "ui/frame";
-import { urbanAirshipSettings } from './urbanAirshipSettings';
+// import { urbanAirshipSettings } from './urbanAirshipSettings';
```
Comment (vip-git) [inline demo/app/activity.android.ts]: if we are not using this anymore its best to remove commented code. If its needed for reference then can be part of documentation.

**Pull request merged**